### PR TITLE
Move bevy_ui dependency under a feature

### LINKED ISSUE #80: Turn input_mocking into a feature
## What problem does this solve?

Currently `leafwing_input_manager` depends on `bevy_ui` because of the `input_mocking` module.

## What solution would you like?

I see several options:

1. Turn `input_mocking` into a feature.
2. Move `input_mocking` into a separate plugin.
3. Add a feature that disables `bevy_ui` dependency for `input_mocking`

Currently used `render` feature is a [group](
https://github.com/bevyengine/bevy/blob/a291b5aaedb4affcb31df2e2e63cb0c665ffb24a/Cargo.toml#L35-L43) that contains unneeded stuff, like `bevy_ui` or `bevy_text`. I was about to submit a PR but noticed that `bevy_ui` is a required dependency for this plugin.

### PR DESCRIPTION
**Pull request by Shatur**
I moved all `bevy_ui` related functionality under a feature, closes #80.

### CODE REVIEW

**Comment by alice-i-cecile** [comment]
Can you retarget this to the `dev` branch? Unfortunately I can't use `main` for active development without eternally confusing users about broken examples.

**Comment by Shatur** [comment]
Oh, sorry, totally forgot about it. Let me rebase the changes.

**Comment by Shatur** [comment]
On `dev` branch you have one doc-test broken before my changes.

**Comment by alice-i-cecile** [comment]
Yep; I was aware that `dev` is currently failing. Thanks for the PR!

File: tests/integration.rs [hunk 1]
```diff
@@ -125,6 +125,7 @@ fn run_in_state() {
 }
 
 #[test]
+#[cfg(feature = "ui")]
```
Comment (alice-i-cecile) [inline tests/integration.rs:128]: Can you please add `bevy_ui` as a dev dependency and remove the feature flag here? Testing in various configurations is generally a pain.

Comment (Shatur) [reply tests/integration.rs:128]: But `ui` is enabled by default. So you will be able to run all tests without changing the configuration.

**Pull request merged**